src/material/table/table-data-source.ts

```diff
@@ -40,6 +40,8 @@ export interface MatTableDataSourcePaginator {
   initialized: Observable<void>;
   pageSize: number;
   length: number;
+  firstPage: () => void;
```

Comment:
This seems like something that needs to be added to the docs / migration guide

Reply:
Yup, definitely - if the paginator is not provided explicitly, it will default to this limited interface. Here's what we have in the docs about it:

"The paginator property of the `MatTableDataSource` has a generic interface that matches most of the paginator API. You may need to explicitly type the paginator to access the full API, e.g. new `MatTableDataSource<MyData, MatPaginator>();`"